src/renderer/graph/FrequencyResponseChart.tsx

```diff
@@ -21,41 +42,131 @@ const FrequencyResponseChart = () => {
     useAquaContext();
   const prevFilters = useRef<IFilter[]>([]);
   const prevFilterLines = useRef<ChartDataPoint[][]>([]);
+  const prevFreqPoints = useRef<ChartDataPointWithId[][]>([]);
 
   const {
     chartData,
+    pointData,
     autoPreAmpValue,
-  }: { chartData: ChartData[]; autoPreAmpValue: number } = useMemo(() => {
+  }: {
+    chartData: ChartCurveData[];
+    pointData: ChartPointsData[];
+    autoPreAmpValue: number;
+  } = useMemo(() => {
+    const freqData = filters.map(({ id, frequency }) => {
+      return { frequency, id };
+    });
+
+    const isFreqDiff =
+      prevFilters.current.length === freqData.length &&
+      !prevFilters.current.every(
+        ({ frequency }, i) => frequency === freqData[i].frequency
+      );
+
+    const hasNewFilter = prevFilters.current.length < freqData.length;
+
+    // Identify the index of each filter by id from previous render
+    const prevIndices: { [id: string]: number } = {};
+    prevFilters.current.forEach((f, i) => {
+      prevIndices[f.id] = i;
+    });
+
     // Update filter lines that have changed
-    const updatedFilterLines = filters.map((f, index) =>
-      index < prevFilters.current.length &&
-      isFilterEqual(f, prevFilters.current[index])
-        ? prevFilterLines.current[index]
-        : getFilterPoints(f)
-    );
+    const { data: updatedFilterLines, points: updatedFreqPoints } = filters
+      .map((f) => {
+        const prevIndex = prevIndices[f.id];
```

Comment:
Now that we use ids, we should take this opportunity to store rendered lines by id. Since we sort the filters, its possible for us to move the left most filter to the right and cause us to rerender all the curve lines.